dht/range_streamer.cc

```diff
@@ -290,20 +290,19 @@ future<> range_streamer::stream_async() {
                 try {
                     for (auto it = range_vec.begin(); it < range_vec.end();) {
                         ranges_to_stream.push_back(*it);
-                        it = range_vec.erase(it);
```

Comment:
Ah, so in an earlier commit you introduce the use-after-move in the calls to `request_ranges` and `transfer_ranges` above, and here you fix it.
Instead, this commit should go first.